### PR TITLE
macos: Add a sidebar icon

### PR DESCRIPTION
**Pull request by jturcotte**
In the process, use an iconset to generate the icns using iconutil.
Also add some missing icon resolutions according to the guidelines.

Issue #296

### CODE REVIEW

**Comment by SamuAlfageme** [comment]
@jturcotte this is dope! 😎 

Will try it later, can you post a screenshot to see how does it look for reference?

**Comment by jturcotte** [comment]
I just picked the same icon we have and put it there, and this is the result:

screen shot 2017-07-31 at 11 22 01

It would be better if some designer would do it but I think it's not too bad already.

File: src/gui/CMakeLists.txt [hunk 1]
```diff
@@ -232,7 +232,7 @@ if (NOT DEFINED APPLICATION_ICON_NAME)
     set(APPLICATION_ICON_NAME ${APPLICATION_SHORTNAME})
 endif()
 
-kde4_add_app_icon( ownCloud "${theme_dir}/colored/${APPLICATION_ICON_NAME}-icon*.png")
+kde4_add_app_icon( ownCloud "${theme_dir}/colored/${APPLICATION_ICON_NAME}-*.png")
```
Comment (guruz) [inline src/gui/CMakeLists.txt]: The files in theme/colored still have the "-icon" in their name, any reason to change this ```kde4_add_app_icon``` thingy?

**Comment by SamuAlfageme** [comment]
@guruz @jturcotte can any of you guys change the PR's target so I can build the installers with rotor (remember the limitations to build fork's PR with it 😕) while keeping master clean 'till I do some good ol' testin'?

**Comment by guruz** [comment]
ACK, @jturcotte please push to this repo, not your own

**Comment by jturcotte** [comment]
I asked for **code review**, the CI reported "All checks have passed", this is not a whole QA process. We do not ask this from all contributors.

@guruz You don't even ask for code reviews and push your stuff directly to master yourself, what is your point?

**Comment by jturcotte** [comment]
I've been under the impression that QA is happening in master, and I think that it's not worth doing custom rotor builds on feature branches to test once, and then test again for any possible merge issue in master, but this is for the QA team to decide. It seems to be the consensus for that reason that the team should make pull request from the main repo only and I'll do it if you really think that this kind of consistency is worth the [proliferating amount of stale branches](https://github.com/owncloud/client/branches/stale), the main reason why I've been avoiding it personally.

I can't retarget the pull request, so for this time I think it should be fine.

I tested the build on my machine and I'm pretty confident it's going to pass on rotor as well. Even if I does break the build, that I add a commit here or on master afterwards won't change anything as long as I push the fix shortly after the merge.

**Comment by guruz** [comment]
@jturcotte i've deleted like 10-15 branches now from that list

**Comment by guruz** [comment]
@SamuAlfageme Build created https://download.owncloud.com/desktop/daily/ownCloud-2.4.0.8114-nightly20170731.pkg

File: src/gui/CMakeLists.txt (hunk 1 above)
Comment (guruz) [inline src/gui/CMakeLists.txt]: ^^

**Comment by guruz** [comment]
@jturcotte great works, it works for me on 10.12.5
However I need to move the Folder manually to sidebar, I'll work on a patch to change this.
https://github.com/owncloud/client/issues/455#issuecomment-291885831

File: src/gui/CMakeLists.txt (hunk 1 above)
Comment (jturcotte) [inline src/gui/CMakeLists.txt]: Yes that's to be able to also get files like `owncloud-sidebar-36.png` in the glob. `-icon` is for the application icon.

Comment (guruz) [inline src/gui/CMakeLists.txt]: Aaah it's because of ```macro (KDE4_ADD_APP_ICON appsources pattern)``` and has nothing to do with kd4...

**Comment by SamuAlfageme** [comment]
@jturcotte is there a way to override the icon to be the default's (i.e. the old one) in case of a branding client with no alternative choice?

**Comment by guruz** [comment]
I think the default won't copy the icon into the bundle if it is not found in the theme. Then the OS will behave as currently

**Comment by jturcotte** [comment]
@michaelstingl Is there any theming documentation I could update to list needed and optional icon sizes with their name?

**Comment by SamuAlfageme** [comment]
@jturcotte we can probably ease @jvillafanez's job updating ownBrander by posting it on https://github.com/owncloud/ownbrander/issues/748 - I think there's no theming docs available

**Comment by SamuAlfageme** [comment]
@jturcotte also, would it be hard to consider a migration path for existing local sync folders to be "updated" in the finder sidebar with the new icon when upgrading the client?

**Comment by guruz** [comment]
> a migration path for existing local sync folders

I'm working on that, there is a bug -> https://github.com/owncloud/client/pull/5928#issuecomment-319307968

**Comment by jancborchardt** [comment]
So cool, nice work @jturcotte!

**Comment by guruz** [comment]
See the linked PR which is related for depending on which wizard you use.

There is no need for a migration path, the Finder will automatically pick up the sidebar icon :)

**Pull request merged**